### PR TITLE
[MNO-324] - Change SuperAdmin role - frontend

### PR DESCRIPTION
**Pull request by clemthenem**
Removed the possibility for Super User to modify their role if they are the only super user left.

### CODE REVIEW

File: src/app/views/marketplace/marketplace-app.controller.coffee [hunk 1]
```diff
@@ -30,6 +30,55 @@ angular.module 'mnoEnterpriseAngular'
...
+                vm.authorized_organizations = {}
+
+                vm.filterAuthorizedOrga(response.organizations)
+            )
+        )
+
+
+      # Filter the authorized organizations for this user
+      vm.filterAuthorizedOrga = (organizations) ->
+        organizations.map(
```
Comment (alexnoox) [inline src/app/views/marketplace/marketplace-app.controller.coffee]: Could you use lodash for the map?

File: src/app/views/marketplace/marketplace-app.controller.coffee [hunk 2]
```diff
@@ -30,6 +30,55 @@ angular.module 'mnoEnterpriseAngular'
...
+
+      vm.changeOrganisation = ->
+        MnoeOrganizations.get(vm.current_organization.id).then(
+          (response) ->
+        )
+
+      vm.addApplication = ->
+        data = {
+          organization_id: vm.current_organization.id
+          app: ["sagelive"]
```
Comment (alexnoox) [inline src/app/views/marketplace/marketplace-app.controller.coffee]: The app to add is hard coded

File: src/app/views/marketplace/marketplace-app.controller.coffee [hunk 3]
```diff
@@ -30,6 +30,55 @@ angular.module 'mnoEnterpriseAngular'
...
+        }
+        MnoeMarketplace.addApp(data)
+        console.log "Wawwwwww"
```
Comment (alexnoox) [inline src/app/views/marketplace/marketplace-app.controller.coffee]: Please remove debug logs

**Comment by clemthenem** [comment]
I accidentally pushed this code far from being finished. I'llreverse the changes so that you can merge the initial work. Will post the rest on a seperate branch of course.

**Comment by clemthenem** [comment]
I added a service that was not defined as a dependency but still used in DashboardOrganizationMember.

File: src/app/views/company/members/organization-members.directive.coffee [hunk 4]
```diff
@@ -34,17 +35,22 @@ DashboardOrganizationMembersCtrl = ($scope, $modal, $sce, MnoeOrganizations, Mno
     MnoeOrganizations.can.create.member()
 
   $scope.isEditShown = (member) ->
-    MnoeOrganizations.can.update.member(member)
+    MnoeOrganizations.can.update.member(member, $scope.hasManySuperAdmin)
```
Comment (alexnoox) [inline src/app/views/company/members/organization-members.directive.coffee]: Could this be more like this?

```
MnoeOrganizations.can.update.member(member) if $scope.hasManySuperAdmin
```

File: src/app/components/mnoe-api/organizations.svc.coffee [hunk 5]
```diff
@@ -12,6 +12,12 @@ angular.module 'mnoEnterpriseAngular'
     @getSelectedId = ->
       _self.selectedId
 
+    # Store the current user role
+    @currentUserRole = null
+
+    @getCurrentUserRole = ->
+      _self.currentUserRole
```
Comment (alexnoox) [inline src/app/components/mnoe-api/organizations.svc.coffee]: Couldn't this method be:
```
@getCurrentUserRole = ->
   $scope.hasManySuperAdmin = _.filter($scope.members, {'role': 'Super Admin'}).length <= 1
```

Comment (alexnoox) [inline src/app/components/mnoe-api/organizations.svc.coffee]: And remove all the `updateUserRole()` calls?

**Comment by alexnoox** [comment]
@ouranos FYI

**Pull request merged**